From 5c1389591a8bf16f2233a47e7c197d0089e17226 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sat, 13 Sep 2014 19:58:16 -0400
Subject: [PATCH] Force Disable Spawn In Memory

Doesn't apply well to our server design and makes the server take longer to spawn.
---
 src/main/java/net/minecraft/server/World.java        | 2 +-
 src/main/java/org/bukkit/craftbukkit/CraftWorld.java | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 4e4af2e..326e39e 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -194,7 +194,7 @@ public abstract class World implements IBlockAccess {
     // CraftBukkit start
     private final CraftWorld world;
     public boolean pvpMode;
-    public boolean keepSpawnInMemory = true;
+    public boolean keepSpawnInMemory = false; // EMC
     public ChunkGenerator generator;
     public final org.spigotmc.SpigotWorldConfig spigotConfig; // Spigot
 
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index b603521..9198bbe 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -1152,6 +1152,7 @@ public class CraftWorld implements World {
     }
 
     public void setKeepSpawnInMemory(boolean keepLoaded) {
+        if (true) return; // EMC
         world.keepSpawnInMemory = keepLoaded;
         // Grab the worlds spawn chunk
         ChunkCoordinates chunkcoordinates = this.world.getSpawn();
-- 
1.9.1

